gateways/checkoutcom/checkoutcom.go

```diff
@@ -0,0 +1,170 @@
+package checkoutcom
+
+import (
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	"github.com/checkout/checkout-sdk-go"
+	"github.com/checkout/checkout-sdk-go/payments"
+	"net/http"
+	"strconv"
+)
+
+// checkoutomClient uses API-Key and custom http client to make http calls
+type CheckoutComClient struct {
+	apiKey     string
+	httpClient *http.Client
+}
+
+const AcceptedStatusCode = 202
+
+// NewClient creates a CheckoutComClient
+// Note: the environment is indicated by the apiKey. See "isSandbox" assignment in checkout.Create.
+func NewClient(apiKey string) *CheckoutComClient {
+	return NewWithHTTPClient(apiKey, common.DefaultHttpClient())
+}
+
+// NewWithHTTPClient uses a custom http client for requests
+func NewWithHTTPClient(apiKey string, httpClient *http.Client) *CheckoutComClient {
+	return &CheckoutComClient{
+		apiKey:     apiKey,
+		httpClient: httpClient,
+	}
+}
+
+// Authorize a transaction for specified amount
+func (client *CheckoutComClient) Authorize(request *sleet.AuthorizationRequest) (*sleet.AuthorizationResponse, error) {
+	config, err := checkout.Create(client.apiKey, nil)
+	if err != nil {
+		return nil, err
+	}
+	config.HTTPClient = client.httpClient
+
+	var checkoutDCClient = payments.NewClient(*config)
```

Comment:
Is this a legacy name from before renaming the package to checkoutcom?

Reply:
Yup, updated